### ISSUE TITLE
Als gebruiker wil ik aan een procesobject een einddatum kunnen toekennen...

### ISSUE THREAD

**Issue by MarcoKlerks** [VNG-Realisatie/gemma-zaken#724]
...zodat o.a. de archiefactiedatum van de zaak berekend kan worden.

De archiefactiedatum is de datum waarop de zaak vernietigd dan wel overgebracht mag worden. Deze wordt berekend zoals beschreven in #345. Bij het berekenen van die datum wordt o.m. de procestermijn (term uit de VNG-selectielijst) gebruikt. De procestermijn hangt af van het procesobject (ook een term uit de VNG-selectielijst). Voorbeelden zijn:
- De bewaartermijn begint te lopen na einde **dienstverband**,
- De bewaartermijn begint te lopen na einde **contract**,
- De bewaartermijn begint te lopen na sloop van het **pand**,
- De bewaartermijn begint te lopen na het verlopen van de **vergunning**.

Op het moment dat de zaak wordt afgehandeld, is het mogelijk nog onbekend wat de einddatum is van (de geldigheid van) het procesobject.  Dit kan zelfs jaren later pas bekend worden.

Op het moment dat de einddatum wordt ingevuld, zal de archiefactiedatum berekend moeten worden. Wanneer de einddatum gewijzigd / gecorrigeerd wordt, moet de archiefactiedatum herrekend worden.

**Bepaling prioriteit door PO**

- [ ] verbreding of verdieping API's
- [ ] stimuleert gebruik door gemeenten
- [ ] stimuleert gebruik door leveranciers

... eventueel nog toelichting door PO

**Definition of ready**
- [ ] Iedereen in het team begrijpt de user story
- [ ] de gewenste (aanvulling op de) functionaliteit van de API's duidelijk en beschreven is.
- [ ] Is klein genoeg (maximaal 1/5 van sprint)
- [ ] Product Owner akkoord en voorzien van prioriteit (mag alleen afgevinkt worden door PO)
- [ ] Idee hebben van hoe deze user story kan worden gedemonstreerd.
- [ ] _**Userstory is voorzien van een analyse (met daarin architectuur, IM, technische beslissingen afgestemd met team)**_
- [ ] Vastgelegd in Github en geplaatst in kolom ready

**Definition of done**
- [ ] Er is een OAS 3.0 specificatie
- [ ] Er is een referentieimplementatie
- [ ] Er zijn tests aanwezig die de wijziging aantonen
- [ ] De technische specificatie (standaard.md) is gepubliceerd leesbaar

**Acceptatiecriteria**
- [ ] De DSO URI- en API-strategie worden gevolgd of afwijkingen zijn vastgelegd als ontwerp keuze
- [ ] Eventueel gemaakte ontwerp keuzes zijn gedocumenteerd
- [ ] Er zijn geen bekende GEMMA tegenstrijdigheden of afwijkingen zijn vastgelegd.

**Taken**
- [ ] Implementeren in referentie-implementatie [_verantwoordelijke_]
- [ ] Schrijven (unit) test voor referentie-implementatie [_verantwoordelijke_]
- [ ] Genereren/opstellen van OAS 3.0 [_verantwoordelijke_]
- [ ] Human Readable publiceren Open API Specificatie (v.3.0) [_verantwoordelijke_]

**Comment by Hugo-ter-Doest** [comment]
Basisfunctionaliteit, dus prio ZH.

**Comment by sergei-maertens** [comment]
Waar leeft de einddatum van het procesobject? Houden we een abstract procesobject met die gegevens bij? Leeft dat in de registratie van het procesobject zelf (concrete invulling)?

**Comment by Hugo-ter-Doest** [comment]
Ja, ik denk dat we eerst een abstract procesobject maken in de ORC. Later wordt de einddatum van het procesobject opgehaald bij het echte procesobject zoals contract of uitkering. Nu leeft de einddatum bij het abstracte procesobject. Er zijn een paar procesobjecten zoals besluit en resultaat die we zelf hebben.

**Comment by joeribekker** [comment]
@MarcoKlerks en @hdksi graag verduidelijking over hoe we dit precies zouden moeten implementeren.

**Comment by hdksi** [comment]
Afspraak gemaakt met @ArjanKloosterboer om me voor eens en altijd te laten uitleggen hoe dit nu in elkaar zit...
